game.cpp

```diff
@@ -1222,31 +1220,22 @@ void test_game() //!OCLINT tests may be many
       eat_nth_food(g,0); // throws exception
     }
     catch ( const std::exception& e ) {
-      assert(e.what() == "You cannot eat food that already has been eaten!");
+      assert(std::string(e.what()) == std::string("You cannot eat food that already has been eaten!"));
     }
   }
-#endif
-
-#define FIX_ISSUE_261
-#ifdef FIX_ISSUE_261
+  // number of food item stays the same,
+  // only the state of food item changes after they are eaten
+  // eaten food items are ?probably removed by game::tick
   {
-    try {
-      game g; //by default one uneaten food
-      int n_food_items_begin = count_food_items(g);
-      assert(has_food(g));
-      eat_nth_food(g,0);
-      assert(!has_food(g));
-      //number of food item stays the same only the state of food item changes after they are eaten
-      assert(n_food_items_begin == count_food_items(g));
-    }
-    catch ( const std::logic_error& e ) {
-      std::cout << e.what();
-    }
+    game g; //by default one uneaten food
+    const int n_food_items_begin = count_food_items(g);
+    assert(has_food(g));
+    eat_nth_food(g,0);
+    assert(!has_food(g));
+    assert(n_food_items_begin == count_food_items(g));
```

Comment:
this still needs to be defined in an issue (Other issue May be, since it is testing different behavioral patterns)